### ISSUE TITLE
Add Location Map to Structure Report

### ISSUE THREAD

**Issue by cstevenson1** [leeensminger/DelDOT-NPDES-Viewer#415]
@KCI-Ablowers @gerrykelly Similar to the location map on BMP Reports (see snip below), DelDOT would like to add a general location map to the Structure Reports. The report feature should be selected on the map. Ideally, I think the location map would go on the first page of the report.

![image](https://user-images.githubusercontent.com/16919898/158867332-d047ef21-c1f9-4811-bc1a-c7f52c9c367f.png)

![image](https://user-images.githubusercontent.com/16919898/158869716-65a4cfb9-d5d5-4948-8f9f-e83cf95b0cf4.png)

A placeholder MXD has been placed at V:\_RESOURCE_MANAGEMENT\Greater_Mid_Atlantic\Tech_Services\LU\DelDot\Reports
If it gets moved, please let us know where to make revisions. The data will also need to be resourced.

**Comment by gerrykelly** [comment]
@cstevenson1 @KCI-Ablowers 
If the map goes on the first page, even if I put the following fields on one line instead of two (maybe not a bad idea anyway)
![image](https://user-images.githubusercontent.com/8259136/159054619-574c41ba-a7f3-4624-9625-eac2a54f9cbb.png)

that only leave room for a map height of about 2.5" vs about 3.5" for the BMP map, so area-wise the map would be only about half the size. 

Options:
- small map (which might be even worse since it could be busier than the BMP maps
- Map on new map only page
- Map on existing 2nd page (Inventory and Inspection Information).  I don't know how much the size of that table varies by structure type, so this is assuming there is always ample space.  If this option selected, above or below table?

**Comment by gerrykelly** [comment]
@cstevenson1 @KCI-Ablowers 
Also, for comparison, here is a map in the report at the 4500 scale of the mxd ..
![image](https://user-images.githubusercontent.com/8259136/159060196-4d55f774-68d2-4c5a-9783-b452f7d6179d.png)

... vs the 1500 that is used for the BMP map:
![image](https://user-images.githubusercontent.com/8259136/159060341-6a2c1e1b-fb66-4c1a-b9b5-68b7484e2614.png)

So, please make a call on scale

**Comment by gerrykelly** [comment]
@cstevenson1 @KCI-Ablowers 
Updated in dev for review.  The map is on the first page as large as can fit

The mxd is located ..
\\deldot-agsdev\d$\Projects\DelDOT_NPDES_Reports\mxds\Structures.mxd
,,, and may be updated in place or replaced with one of the same name

**Comment by cstevenson1** [comment]
@gerrykelly Bruce agreed it is fine on first page but trying to decide how we want to modify the appearance of the selected feature. Using the default selection color can blend with the swale point symbology. Let us know if you have suggestions, otherwise, we'll brainstorm internally.

![image](https://user-images.githubusercontent.com/16919898/159278805-bf2008b7-c18b-406b-81b1-4741c7c4d0f8.png)

**Comment by gerrykelly** [comment]
@cstevenson1 @KCI-Ablowers 
While I could write code to dynamically put a graphic on the mxd, or the jpeg, that would take some time (I think more than it's worth).  What might work well is something like I did for the DNREC BMP on the map, where there is a separate layer where the report code applies a definition query for only the selected feature (and we then don't have a selection in the map, so the "selected" feature is only identified by the symbology for the special one-feature-only-layer).  
So, basically you would copy the existing structures layer and revise the symbology -- maybe same symbols but bigger, maybe red-circle outlines

**Comment by cstevenson1** [comment]
@gerrykelly I think I follow you. Could we put a halo of some sort around the feature in the copied layer similar to the snip below (no callout boxes)?

![image](https://user-images.githubusercontent.com/16919898/159337400-b0bb44de-5727-4067-9d3b-2b7d2278f9b4.png)

**Comment by gerrykelly** [comment]
@cstevenson1 @KCI-Ablowers 
For the sake of getting the code in place and testing, I added a second structures layer to the mxd and have inlets a special symbol in that layer.  I no longer show selected feature symbology in the map, but rather show only the "selected" feature in the top layer.  You can run the report for an inlet and get the idea,  Then update symbology for all in the top layer -- however you'd like -- and see if you like the approach.  If it would work better to have the special-symbol layer lower in the order, and/or apply a definition query to the main structures layer to show all but the selected feature, let me know,

**Comment by cstevenson1** [comment]
@KCI-Ablowers @gerrykelly updated MXD in provided location (updated selected feature symbology, added underdrain features, conveyance drawing order, etc.). Matt and Bruce approved the updated. Please push to prod

We are aware that if the comments are long, it will push the map to the second page as shown below but we decided this is fine for now since 90+% of them should fall on the first page.

Updated
![image](https://user-images.githubusercontent.com/16919898/162479534-447953b7-609d-4e2d-8465-990d695e603f.png)


Second page spillover
![image](https://user-images.githubusercontent.com/16919898/162479479-93d25b33-ede2-483b-a888-caec2f9d79ad.png)

**Comment by gerrykelly** [comment]
@cstevenson1 @KCI-Ablowers 
Note: as part of Push to Prod I copied the mxd to the comparable location on the Prod server -- deldot-ags2016 -- and updated the data sources to use the prod GDB.  (always good to verify that tho) any further updates on symbology should be done in place on the mxd

**Comment by gerrykelly** [comment]
@cstevenson1 @KCI-Ablowers
Done (and removing my assignment). Please review.  I thought I was going to need to do a followup update to get the maps on the reports generated by the https://newarkgis.kci.com/arcgis/rest/services/DelDotReports/multiReportStructConvey/GPServer/multiReportStructConvey/submitJob task, but that just seems to be working correctly as-is, (Verify sometime when you're working with it later)

**Comment by cstevenson1** [comment]
@gerrykelly @KCI-Ablowers appears to be working in prod. will create new issue as needed for future enhancements regarding symbology
